lib/SimpleSAML/Metadata/SAMLBuilder.php

```diff
@@ -381,6 +381,9 @@ private static function createEndpoints(array $endpoints, bool $indexed): array
                     }
 
                     $ep['index'] = $maxIndex + 1;
+                    if (isset($ep['isDefault'])) {
```

Comment:
This should also work if no index is explicitly set, so this should go down a few lines, underneath the `$t->setIndex()`